src/App.tsx

```
@@ -8,53 +8,185 @@ import { PostsList } from './components/PostsList';
+import { getPosts } from './Api/Posts';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { createComments, deleteComments, getComments } from './Api/Comment';
+import { Comment } from './types/Comment';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const [isLoading, setIsLoading] = useState(false);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
 
-              <Loader />
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post>();
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const [error, setError] = useState(false);
+  const [errorComments, setErrorComments] = useState(false);
+
+  const [isUserSelected, setIsUserSelected] = useState(false);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+
+  const [comments, setComments] = useState<Comment[]>([]);
+
+  async function fetchCreateComment({
+    name,
+    email,
+    body,
+    postId,
+  }: Omit<Comment, 'id'>) {
+    try {
+      const result = await createComments({ name, email, body, postId });
+
+      setComments(currentComments => [...currentComments, result]);
+    } catch (errorCreatComments) {
```

This catch rethrows the caught error (`throw errorCreatComments;`) without logging or setting any error state. The previous review requested: "Replace empty catch blocks with at least `console.error(error)` and, where appropriate, set error state. Avoid duplicated loading-state resets (e.g., don’t call `setIsLoading(false)` in both catch and finally; keep it in finally only)." Consider replacing the rethrow with `console.error(errorCreatComments)` and either set an error state that the UI can render or return/reject in a controlled way so errors do not become unhandled promise rejections.